src/util/union_find.h

```diff
@@ -129,11 +129,18 @@ class unsigned_union_find
 
 template <typename T>
 // NOLINTNEXTLINE(readability/identifiers)
-class union_find:public numbering<T>
+class union_find final
 {
+  using numbering_typet=numbering<T>;
+  numbering_typet numbering_;
+
+  using number_type=typename numbering_typet::number_type;
+
 public:
   // NOLINTNEXTLINE(readability/identifiers)
-  typedef typename numbering<T>::size_type size_type;
+  using size_type=typename numbering_typet::size_type;
+  using iterator=typename numbering_typet::iterator;
+  using const_iterator=typename numbering_typet::const_iterator;
```

Comment:
As you've changed most of the other `using` to `typedef`: would you mind doing that here as well? (And the "_" in `numbering_` isn't clear to me.)